### ISSUE TITLE
When there is only one label, how can I set it to fill the entire bar instead of a fixed size and then shrink as the number of labels increases?

### ISSUE THREAD

**Issue by linux36524** [Alexays/Waybar#3738]
When there is only one label, how can I set it to fill the entire bar instead of a fixed size and then shrink as the number of labels increases? Tint2 can be set like this, but I don't know how to set waybar.

**Comment by Alexays** [comment]
Yes! Add `"expand": true` to the module. Any module with `expand` enabled is packed into the bar with GTK's expand+fill flags, so it grows to consume all the leftover space, and when several expanding modules share a section they split the space and shrink as you add more:

```jsonc
"custom/label": {
    "format": "{}",
    "expand": true
}
```

You can see it in the source (`src/bar.cpp`): each module is packed with `module->expandEnabled()` as both the expand and fill argument. It's documented per-module in the man pages too (e.g. `man waybar-custom`, look for `expand`).

Hope that does what you're after. Feel free to reopen if not 🙏
